sale_order_secondary_unit/models/sale_order.py

```diff
@@ -11,52 +12,44 @@ class SaleOrderLine(models.Model):
         "uom_field": "product_uom",
     }
 
-    secondary_uom_unit_price = fields.Float(
+    secondary_uom_price_unit = fields.Float(
         string="2nd unit price",
         digits="Product Price",
-        compute="_compute_secondary_uom_unit_price",
+        compute="_compute_secondary_uom_price_unit",
+        store=True,
+        precompute=True,
     )
 
-    product_uom_qty = fields.Float(
-        store=True, readonly=False, compute="_compute_product_uom_qty", copy=True
-    )
-
-    @api.depends("secondary_uom_qty", "secondary_uom_id", "product_uom_qty")
+    @api.depends("secondary_uom_qty", "secondary_uom_id")
     def _compute_product_uom_qty(self):
+        res = super()._compute_product_uom_qty()
         self._compute_helper_target_field_qty()
+        return res
 
     @api.onchange("product_uom")
-    def onchange_product_uom_for_secondary(self):
+    def _onchange_product_uom(self):
```

Comment:
If everything is now computed writable fields, you should convert this as well. If not, you don't know which one is going to act first (the compute or the onchange).

Reply:
[xf](https://github.com/OCA/product-attribute/blob/16.0/product_secondary_unit/models/product_secondary_unit_mixin.py#L23-L25)

I'm not sure that's an option - write does not replace onchange in any way

Reply:
I'm not sure we're on the same page. It's literally implemented as product_secondary_unit in v16 asks for - and it works as a result.

As a side note - if it's computed, so what if there's an action to be performed on onchange?

Reply:
The current implementation of product_secondary_unit is not correct. As the code in the compute method is duplicated in onchange...

I think the migration should have merged both in a compute method.

Reply:
Well, yet with current merged implementation of product_secondary_unit I don't see other way for this to work. Yet as far as I can see, you've approved that in https://github.com/OCA/product-attribute/pull/1325 ? I take it your position has changed since then?

Reply:
We cannot see all the changes effects at each time :sweat_smile: 

Nevertheless, my question is still open. Why is the computed method does not work ?

Reply:
I don't have answer to your question, neither I don't think it's relevant - it works now and it didn't work in another migration. In the end my only interest is having a properly working module, rather then discovering why when following directions provided by product_secondary_unit module it actually works 🤷‍♂️

Reply:
@rousseldenis do you have any does that answer your question?